library/src/amd_detail/hipblas.cpp

```diff
@@ -15522,17 +15522,19 @@ try
 {
     if(info == NULL)
         return HIPBLAS_STATUS_INVALID_VALUE;
+    else if(trans != HIPBLAS_OP_N && trans != HIPBLAS_OP_T && trans != HIPBLAS_OP_C)
+        *info = -1;
     else if(n < 0)
         *info = -2;
     else if(nrhs < 0)
         *info = -3;
-    else if(A == NULL)
+    else if(A == NULL && n)
         *info = -4;
     else if(lda < std::max(1, n))
         *info = -5;
-    else if(ipiv == NULL)
+    else if(ipiv == NULL && n)
         *info = -6;
-    else if(B == NULL)
+    else if(B == NULL && n * nrhs)
         *info = -7;
     else if(ldb < std::max(1, n))
         *info = -8;
```

Comment:
See previous comment. Since the argument index is different for each function not sure how we could get rid of these cleanly, could add comments instead?